### PR TITLE
enh: add version check as a function to etelemetry

### PR DESCRIPTION
**Pull request by satra**

### CODE REVIEW

**Comment by satra** [comment]
@yarikoptic and @mgxd - please take a look.

File: etelemetry/client.py [hunk 1]
```diff
@@ -42,3 +43,69 @@ def get_project(repo, **rargs):
...
+
+    # TODO: run in a separate thread
+    try:
+        import etelemetry
+    except ImportError as exc:
+        lgr.debug(
+            "Cannot check latest available version for %s: %s",
+            project, exc
+        )
+        return
```
Comment (mgxd) [inline etelemetry/client.py]: I'm confused here - is this importing itself?

File: etelemetry/client.py [hunk 2]
```diff
@@ -42,3 +43,69 @@ def get_project(repo, **rargs):
...
+
+    from pkg_resources import parse_version
+
+    latest = {"version": "Unknown", "bad_versions": []}
+    ret = None
+    try:
+        ret = etelemetry.get_project(project)
+    except Exception as e:
+        lgr.debug("Could not check %s for version updates: %s", project, e)
+        return None
```
[the remark this answers is not in the document]
Comment (satra) [reply etelemetry/client.py]: `raise_exception` was for bad versions

**Comment by satra** [comment]
@mgxd - thanks for feedback. if things look ok, do you want to merge and release a new version?

File: etelemetry/client.py [hunk 3]
```diff
@@ -52,6 +53,12 @@ def check_available_version(project, version, lgr=None, raise_exception=False):
     ----------
     project: str
       as on GitHub (e.g., sensein/etelemetry-client. Releases will be checked
+    version: str
+      local version of project
+    lgr: python logger object
+      external logger to be used
+    raise_exception: bool
+      raise an exception is a bad local version is detected
```
Comment (mgxd) [inline etelemetry/client.py]:
```suggestion
      raise an exception if a bad local version is detected
```

**Pull request merged**